### PR TITLE
Allow explicit tls1.0, tls1.1, tls1.2 for startTLS

### LINKED ISSUE #933: [Bug] Swiftm. 6.x only support tls 1.0
### PHP constant change
STREAM_CRYPTO_METHOD_TLS_CLIENT changed from tls1.x to only tls1.0
Reference: https://secure.php.net/manual/de/function.stream-socket-enable-crypto.php#119122

### Line of code
https://github.com/swiftmailer/swiftmailer/blob/master/lib/classes/Swift/Transport/StreamBuffer.php#L94

### Solution
Set option to `STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT|STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT|STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT`

Better: make it changeable than hard coded

### PR DESCRIPTION
**Pull request by Rotzbua**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| Doc update?   | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #933 #598 #689
| License       | MIT

Swiftmailer only allow tls1.0 connections for startTLS
To support modern tls we allow explicit tls1.0, tls1.1, tls1.2

### CODE REVIEW

**Comment by Rotzbua** [comment]
@fabpot This patch also for 6.1?

File: lib/classes/Swift/Transport/StreamBuffer.php [hunk 1]
```diff
@@ -87,7 +87,11 @@ public function setParam($param, $value)
 
     public function startTLS()
     {
-        return stream_socket_enable_crypto($this->stream, true, STREAM_CRYPTO_METHOD_TLS_CLIENT);
+        // STREAM_CRYPTO_METHOD_TLS_CLIENT only allow tls1.0 connections (some php versions) 
+        // To support modern tls we allow explicit tls1.0, tls1.1, tls1.2
+        // Ssl3 and older are not allowed because they are vulnerable
+        // @TODO make tls arguments configurable
+        return stream_socket_enable_crypto($this->stream, true, STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT);
```
Comment (fabpot) [inline lib/classes/Swift/Transport/StreamBuffer.php]: The second argument of `stream_socket_enable_crypto` is now optional. In that case, it takes whatever was configured on the stream context. So, wouldn't it be "better" to remove the second argument here, set a default on the stream when it's created and let the user configure it via `setParam()` like other stream context options?

Comment (Rotzbua) [reply lib/classes/Swift/Transport/StreamBuffer.php]: I think `stream_socket_enable_crypto` overwrites the settings of steam context, right?
So `stream_socket_enable_crypto($this->stream, true);` would be the correct code if we would use `setParms`. But then a documentation about enabeling tls1.1/1.2 for php7.0 and 7.1 should be added and thats a bigger patch and not this fast fix..

BTW with php7.2 they reverted STREAM_CRYPTO_METHOD_TLS_CLIENT to all versions of tls1.X, so the problem is only in php7.0 and 7.1

**Comment by ben-crowhurst** [comment]
Hey All 👋
How can I help to get this over the line?  Is there further development required? if so spell it out and I'll get it done.

**Comment by fabpot** [comment]
I messed up with this PR (or the tool I use messed up). I had to reconstrcut the commit, sorry about that. But the code is on master now.

**Pull request closed without merging**